src/config.js

```diff
@@ -0,0 +1,181 @@
+   * @property {string?} payloadDelimiter - Seperators of nested attributes.
+   * @property {string?} payloadFilePath - Location of a JSON request payload.
+   * @property {boolean} payloadTemplated - If templated values are replaced.
+   * @property {string?} proxy - An optional proxied connection for requests.
+   * @property {Retries} retries - The retries method to use for failed requests.
+   * @property {string?} token - The authentication value used with the Slack API.
+   * @property {string?} webhook - A location for posting request payloads.
+   * @property {string?} webhookType - Posting method to use with the webhook.
+   */
+
+  /**
+   * @type {Inputs} - The actual action input values.
+   */
+  inputs;
+
+  /**
+   * @type {import("axios").AxiosStatic} - The axios client.
+   */
+  axios;
+
+  /**
+   * @type {Content} - The parsed payload data to send.
+   */
+  content;
+
+  /**
+   * Shared utilities specific to the GitHub action workflow.
+   * @type {import("@actions/core")}
+   */
+  core;
+
+  /**
+   * @type {import("@slack/web-api")} - Slack API client.
+   */
+  webapi;
+
+  /**
+   * Gather values from the job inputs and use defaults or error for the missing
+   * ones.
+   *
+   * The content of the payload is also parsed, proxies set, and a shared "core"
+   * kept for later use.
+   *
+   * @constructor
+   * @param {core} core - GitHub Actions core utilities.
+   */
+  constructor(core) {
+    this.axios = axios;
+    this.core = core;
+    this.webapi = webapi;
+    this.inputs = {
+      errors: core.getBooleanInput("errors"),
+      method: core.getInput("method"),
+      payload: core.getInput("payload"),
+      payloadDelimiter: core.getInput("payload-delimiter"),
+      payloadFilePath: core.getInput("payload-file-path"),
+      payloadTemplated: core.getBooleanInput("payload-templated") || false,
+      proxy:
+        core.getInput("proxy") ||
+        process.env.HTTPS_PROXY ||
+        process.env.https_proxy ||
+        null,
+      retries: core.getInput("retries") || this.Retries.FIVE,
+      token: core.getInput("token") || process.env.SLACK_TOKEN || null,
+      webhook:
+        core.getInput("webhook") || process.env.SLACK_WEBHOOK_URL || null,
+      webhookType: core.getInput("webhook-type"),
+    };
+    this.validate();
+    core.debug(`Gathered action inputs: ${JSON.stringify(this.inputs)}`);
+    this.content = new Content().get(this);
+    core.debug(`Parsed request content: ${JSON.stringify(this.content)}`);
+  }
+
+  /**
+   * Confirm the configurations are correct enough to continue.
+   */
+  validate() {
+    switch (this.inputs.retries) {
+      case this.Retries.ZERO:
+      case this.Retries.FIVE:
+      case this.Retries.TEN:
+      case this.Retries.RAPID:
+        break;
+      default:
+        core.warning(
+          `Invalid input! An unknown "retries" value was used: ${this.inputs.retries}`,
+        );
+    }
+    switch (true) {
```

Comment:
😳 `switch (true)` ? How .. does that work? And also how does that combine with some of the `case`s not having a `break`?

Reply:
😱 This is a hacky section... The `switch (true)` is checking for a `true` condition and is similar to just a regular `if` with the following cases being an `else if`, although unlike a regular `if` block, these cases can fall through or [be true multiple times](https://github.com/slackapi/slack-github-action/blob/5ae8b010cbd83442a9dde4b2fc077ceffd17a61f/src/config.js#L123-L133)! 👀

The cases that always throw a `SlackError` avoid including a `break`, while the cases that are along the happy path will `break` at the end. This is not super clear 😬

I'm finding this pattern a few times throughout other configurations and realize it's unclear, especially with the `!!this.inputs.example` conditions to check for **some** value, so please let me know if a refactoring is in order 🙏

Reply:
Eh if it works it's fine.. was just 🤯 is all. Thanks for explaining 🙇

Reply:
Haha it takes me a second to remember what a `true` switch even does... Wild booleans to find all around 🌵